src/app/public/src/modules/action-buttons/action-buttons.component.ts

```diff
@@ -1,13 +1,42 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnInit } from '@angular/core';
 
 import { StacheNav, StacheNavLink, StacheNavComponent } from '../nav';
+import { InputConverter } from '../shared';
 
 @Component({
   selector: 'stache-action-buttons',
   templateUrl: './action-buttons.component.html',
   styleUrls: ['./action-buttons.component.scss']
 })
-export class StacheActionButtonsComponent extends StacheNavComponent implements StacheNav {
+export class StacheActionButtonsComponent extends StacheNavComponent implements StacheNav, OnInit {
   @Input()
   public routes: StacheNavLink[];
+
+  @Input()
+  @InputConverter()
+  public showSearch: boolean = true;
+
+  private originalRoutes: StacheNavLink[];
+
+  public ngOnInit() {
+    this.originalRoutes = this.routes.slice();
+  }
+
+  public searchApplied(searchText: string) {
+    let filteredItems = this.originalRoutes;
+    if (searchText) {
+      filteredItems = this.routes.filter(function (item: any){
```

Comment:
`this.originalRoutes.filter`     Filtering against the routes causes an issue when you try to search a couple times, it continues to search against the filtered list. 
Search for 2 ->   search for 3 ->   list disappears completely.